e2e/auth/login.testSuite.ts

```diff
@@ -218,99 +218,141 @@ export const authLoginTestSuite = () => {
-      );
-      return request(app.getHttpServer())
-        .get('/v1/search/test')
-        .set('Cookie', [
-          loginResponse.headers['set-cookie'][0].split(';')[0] + ';',
-          loginResponse.headers['set-cookie'][1].split(';')[0] + ';',
-        ])
-        .expect(200);
-    });
+    describe('POST /login/siwe/verify', () => {
+      let message: string;
+      let signature: string;
+      let loginResponse: request.Response;
 
-    it('same nonce can not be used for two logins', async () => {
-      const { text } = await request(app.getHttpServer())
-        .post('/v1/login/siwe/initiate')
-        .expect(201);
-      const nonce = JSON.parse(text).nonce;
-      const { message, signature } = await signSiweMessage(nonce);
-
-      const loginResponses = await Promise.all([
-        request(app.getHttpServer()).post('/v1/login/siwe/verify').send({
-          message,
-          signature,
-        }),
-        request(app.getHttpServer()).post('/v1/login/siwe/verify').send({
-          message,
-          signature,
-        }),
-      ]);
-
-      expect(loginResponses.filter((r) => r.status === 201).length).toBe(1);
-      expect(loginResponses.filter((r) => r.status === 401).length).toBe(1);
-    });
+      describe('when SIWE message is valid', () => {
+        describe('when login origin is set', () => {
+          beforeAll(async () => {
+            ({ loginResponse } = await loginWithSiwe(wallet, origin));
+          });
+
+          it('login response should contain access and refresh tokens', () => {
+            expect(loginResponse.status).toBe(201);
+
+            expect(loginResponse.headers['set-cookie']).toHaveLength(2);
+            expect(loginResponse.headers['set-cookie']).toEqual(
+              expect.arrayContaining([
+                expect.stringMatching(
+                  /(?:token|refreshToken)=.+\..+\..+; Path=\/; HttpOnly; Secure; SameSite=None/
+                ),
+              ])
+            );
+            expect(loginResponse.headers['set-cookie'][0]).toContain(
+              loginResponse.body.token
+            );
+            expect(loginResponse.headers['set-cookie'][1]).toContain(
+              loginResponse.body.refreshToken
+            );
+          });
+
+          it.todo(
+            'should not be authorized when request origin is not specified'
+          );
+
+          it('should be authorized when request origin matches origin of access token', async () => {
+            return request(app.getHttpServer())
+              .get('/v1/search/test')
+              .set('Cookie', [
+                loginResponse.headers['set-cookie'][0].split(';')[0] + ';',
+                loginResponse.headers['set-cookie'][1].split(';')[0] + ';',
+              ])
+              .set('Origin', origin)
+              .expect(200);
+          });
+
+          it('should not be authorized when request origin does not matches origin of access token', async () => {
```

Comment:
maybe specifying what should technically happen would be better? This description is ambiguous.

Reply:
Yes, I am personally prefer terse descriptions. Full version would be 
`Authorized endpoint should return Unauthorized exception, when request origin does not match origin of acees token`

Does this sounds more precise from your perspective?

Reply:
why not describe what actually is checked? In this case response status code, not exception. It really helps when something fails in tests and it is possible to see what was wrong without looking into the test.

Reply:
what about: `when request origin does not match access token origin, it should respond with `401` status code?
this is what actually is tested, not the exception returned.

Reply:
Ok, sounds good